### PR TITLE
Cdk bridge helper

### PR DESCRIPTION
**Pull request by Eikix**
Exposes a helper `getCdkProperty` that helps resolve user's cdk resources attribute names inside Serverless framework config files.

Example: 
```
import { AWS } from '@serverless/typescript';

import { getCdkProperty } from 'helper';

import { MyConstruct } from './myConstruct';

const serverlessConfiguration: AWS = {
  service: 'test-app',
  configValidationMode: 'error',
  plugins: ['../../src'],
  provider: {
    name: 'aws',
  },
  functions: {
    testFunction: {
      environment: {
        ORCHESTRATOR_TABLE_NAME: getCdkProperty<MyConstruct>('dynamodbName'),
        ORCHESTRATOR_TABLE_ARN: getCdkProperty<MyConstruct>('dynamodbArn'),
      },
      handler: './lambda.js',
    },
  },
  custom: {
    myConstruct: MyConstruct,
  },
};

module.exports = serverlessConfiguration;


```

### CODE REVIEW

**Comment by CorentinDoue** [comment]
C'est bizarre ton commit de merge à la fin:

- soit tu fais deux PRs qu'on merge séparément car la doc à pas besoin du helpers et le helpers a pas besoin de la doc
- soit tu cherry-pick proprement ton commit de doc si tu veux que tout passe en même temps

File: packages/serverless-cdk-plugin/package.json [hunk 1]
```diff
@@ -63,9 +61,15 @@
...
     "ts-node": "^10.8.1",
+    "ts-toolbelt": "^9.6.0",
     "ttypescript": "^1.5.13",
     "typescript": "^4.7.4"
   },
+  "exports": {
+    "./helpers": "./dist/cjs/helpers.js",
+    ".": "./dist/cjs/index.js"
+  },
+  "moduleResolution": "Node16",
```
Comment (fargito) [inline packages/serverless-cdk-plugin/package.json:72]: Are these necessary?

File: packages/serverless-cdk-plugin/package.json [hunk 2]
```diff
@@ -74,5 +78,12 @@
         ]
       }
     }
+  },
+  "typesVersions": {
+    "*": {
+      "helpers": [
+        "./dist/types/helpers.d.ts"
+      ]
+    }
```
Comment (fargito) [inline packages/serverless-cdk-plugin/package.json:87]: Is this necessary?

File: packages/serverless-cdk-plugin/package.json (hunk 1 above)
Comment (Eikix) [reply packages/serverless-cdk-plugin/package.json:72]: We want our plugin to export two components: the serverless plugin as well as a helper function to facilitate autocomplete for construct attributes.

We were unable to export both the serverless plugin and a helper function in our `index.ts `:

This works:
```ts
import { ServerlessCdkPlugin } from './serverlessCdk';

module.exports = ServerlessCdkPlugin;
```

While this does not work:
```ts
import { ServerlessCdkPlugin } from './serverlessCdk';
import { getCdkProperty } from './helper';

module.exports = {ServerlessCdkPlugin, getCdkProperty};
```

We found that this exotic approach solves our issue.

File: packages/serverless-cdk-plugin/package.json (hunk 2 above)
Comment (Eikix) [reply packages/serverless-cdk-plugin/package.json:87]: This part is the continuation of our atypical method(:

**Review by CorentinDoue — APPROVED** [review]

**Review by GuillaumeLagrange — APPROVED** [review]

**Comment by GuillaumeLagrange** [comment]
I suggest we close this and continue the discussion in #207

**Comment by GuillaumeLagrange** [comment]
closed and continued in #213

**Pull request closed without merging**